### PR TITLE
Misc items for optim PR

### PR DESCRIPTION
**Pull request by GiudGiud**
- doc on transfers is outdated now that it uses siblings
- adjust parsed parameters since it s a little unfair to make you guys adjust to my unmerged PR
- use functors more, that part is WIP but fairly simple

### CODE REVIEW

**Comment by GiudGiud** [comment]
@zachmprince 
the inconvenient part of FunctorElementalGradientAux is that because the functor is a variable (so it s AD), naturally we d want the material property to be AD too. Does that work with you?

@lindsayad any objection to this hybrid object with both functors and regular material properties?

**Comment by lindsayad** [comment]
> @lindsayad any objection to this hybrid object with both functors and regular material properties?

It looks a little wonky. You don't want the factor to be a functor too?

**Comment by GiudGiud** [comment]
Well the material property factor is what they need here. On paper yes I would want a functor factor and in fact there is one as well.

File: framework/src/auxkernels/FunctorElementalGradientAux.C
Comment (GiudGiud) [inline framework/src/auxkernels/FunctorElementalGradientAux.C]: umm forgot an if there

**Comment by dschwen** [comment]
> the factor to be a functor

DependentWideeyedIchidna-size_restricted

File: framework/src/auxkernels/FunctorElementalGradientAux.C
Comment (GiudGiud) [reply framework/src/auxkernels/FunctorElementalGradientAux.C]: no I didnt

File: framework/src/auxkernels/FunctorElementalGradientAux.C
Comment (lindsayad) [inline framework/src/auxkernels/FunctorElementalGradientAux.C]: I *think* you should also be able to do a default value of 1 here as well like you do for the functor so that you don't have to do ternary conditions everywhere

**Comment by lindsayad** [comment]
> On paper yes I would want a functor factor and in fact there is one as well.

Well in fact you do! Haha. Yea I think having the traditional mat prop option is fine then if that's what they need

File: framework/src/auxkernels/FunctorElementalGradientAux.C
Comment (GiudGiud) [reply framework/src/auxkernels/FunctorElementalGradientAux.C]: yeah just tried and it works. I didnt know matprops could be substituted like this.
I guess everything can! functor, functions, variables, matprops.

**Comment by GiudGiud** [comment]
@lindsayad @zachmprince do you want to review this? I believe @dschwen is waiting on your approval of my changes to merge. 

We ll need an independent reviewer on these commits on the MOOSE side anyway so might as well have a look now

File: framework/doc/content/source/auxkernels/FunctorElementalGradientAux.md
Comment (lindsayad) [inline framework/doc/content/source/auxkernels/FunctorElementalGradientAux.md:20]: I didn't know you could do listings from modules in framework doc

**Review by lindsayad — APPROVED** [review]

File: framework/doc/content/source/auxkernels/FunctorElementalGradientAux.md
Comment (zachmprince) [reply framework/doc/content/source/auxkernels/FunctorElementalGradientAux.md:20]: This is a little dangerous since trying to build from `framework/doc/` will cause a failure. I think you can use the [IfElse extension](https://mooseframework.inl.gov/python/MooseDocs/extensions/ifelse.html) to avoid this.

Comment (GiudGiud) [reply framework/doc/content/source/auxkernels/FunctorElementalGradientAux.md:20]: there's over 40 instances of that in framework/doc/content

File: modules/optimization/src/functions/ParsedOptimizationFunction.C
Comment (zachmprince) [inline modules/optimization/src/functions/ParsedOptimizationFunction.C]: Do you think that this would be confusing with the `param_vector_name`? Maybe `param_symbol_names`?

File: modules/optimization/src/functions/ParsedOptimizationFunction.C
Comment (zachmprince) [inline modules/optimization/src/functions/ParsedOptimizationFunction.C]: Maybe `constant_symbol_names`? This isn't quite identical to `ParsedFunction` since it only accepts `Real` values at input.

**Review by zachmprince — COMMENTED** [review]
Suggestions for the changed parameters in ParsedOptimizationFunction

File: framework/doc/content/source/auxkernels/FunctorElementalGradientAux.md
Comment (zachmprince) [reply framework/doc/content/source/auxkernels/FunctorElementalGradientAux.md:20]: Interesting, I guess the the `git_ls_files` works all over the repo.

File: modules/optimization/src/functions/ParsedOptimizationFunction.C
Comment (GiudGiud) [reply modules/optimization/src/functions/ParsedOptimizationFunction.C]: constant_names then?

File: modules/optimization/src/functions/ParsedOptimizationFunction.C
Comment (GiudGiud) [reply modules/optimization/src/functions/ParsedOptimizationFunction.C]: still better than vars
do you prefer that?

Comment (zachmprince) [reply modules/optimization/src/functions/ParsedOptimizationFunction.C]: Yes. It will help me to remember what it is.

File: modules/optimization/src/functions/ParsedOptimizationFunction.C
Comment (zachmprince) [reply modules/optimization/src/functions/ParsedOptimizationFunction.C]: I think `constant_symbol_names` works for me.

**Comment by GiudGiud** [comment]
should be addressed

**Review by zachmprince — APPROVED** [review]

**Comment by GiudGiud** [comment]
@dschwen merge please
(please do not interrupt your vacation for this though)

**Pull request merged**
